src/sanitize.ts

```diff
@@ -0,0 +1,18 @@
+import { removeXrayTraceId } from './middlewares/remove-xray-trace-id';
+import { BaseFetchOptions } from './types';
+
+export function sanitizeOutputs<T>(
+  output: T[],
+  options: BaseFetchOptions
+): T[] {
+  return output.map((i) => {
+    return sanitizeOutput(i, options);
+  });
+}
+
+export function sanitizeOutput<T>(output: T, options: BaseFetchOptions): T {
+  let sanitized = { ...output };
+
+  sanitized = removeXrayTraceId(sanitized, options.extractXrayTrace);
```

Comment:
Could we check here (if options.extractXrayTrace) instead of passing it down the chain?

Reply:
I guess it makes it more clear here, but is it logical that the check if a middleware should be used is outside of the middleware itself? Maybe it is, just double checking

Reply:
To me it seems it's up to sanitize/enrich is the "middlewares runner" and thus decide. Thats my interpretation at least! I guess the middleware could get other options passed to it if we add more advanced ones (lets hope we don't) 😄